snippets/rust.json

```diff
@@ -1,33 +1,446 @@
         "prefix": "unreachable",
         "body": [
-            "unreachable!()"
+            "unreachable!(${1})"
         ],
-        "description": "Insert unreachable!"
+        "description": "unreachable!(…)"
     },
-    "println": {
-        "prefix": "println",
+    "vec": {
+        "prefix": "vec",
         "body": [
-            "println!(\"${1:{\\}}\", $0)"
+            "vec![${1}]"
         ],
-        "description": "Insert println!"
+        "description": "vec![…]"
+    },
+    "write": {
+        "prefix": "write",
+        "body": [
+            "write!(${1}, \"${2}\")"
+        ],
+        "description": "write!(…)"
+    },
+    "writeln": {
+        "prefix": "writeln",
+        "body": [
+            "writeln!(${1}, \"${2}\")"
+        ],
+        "description": "writeln!(…, …)"
+    },
+    "Err": {
+        "prefix": "Err",
+        "body": [
+            "Err(${1})"
+        ],
+        "description": "Err(…)"
+    },
+    "Ok": {
+        "prefix": "Ok",
+        "body": [
+            "Ok(${1:result})"
+        ],
+        "description": "Ok(…)"
+    },
+    "Some": {
+        "prefix": "Some",
+        "body": [
+            "Some(${1})"
+        ],
+        "description": "Some(…)"
+    },
+    "assert": {
+        "prefix": "assert",
+        "body": [
+            "assert!(${1});"
+        ],
+        "description": "assert!(…);"
+    },
+    "assert_eq": {
+        "prefix": "assert_eq",
+        "body": [
+            "assert_eq!(${1}, ${2});"
+        ],
+        "description": "assert_eq!(…, …);"
+    },
+    "bench": {
+        "prefix": "bench",
+        "body": [
+            "#[bench]",
+            "fn ${1:name}(b: &mut test::Bencher) {",
+            "    ${2:b.iter(|| ${3:/* benchmark code */})}",
+            "}"
+        ],
+        "description": "#[bench]"
+    },
+    "const": {
+        "prefix": "const",
+        "body": [
+            "const ${1:CONST}: ${2:Type} = ${4:init};"
+        ],
+        "description": "const …: … = …;"
+    },
+    "derive": {
+        "prefix": "derive",
+        "body": [
+            "#[derive(${1})]"
+        ],
+        "description": "#[derive(…)]"
+    },
+    "else": {
+        "prefix": "else",
+        "body": [
+            "else {",
+            "    ${1:unimplemented!();}",
+            "}"
+        ],
+        "description": "else { … }"
+    },
+    "enum": {
+        "prefix": "enum",
+        "body": [
+            "#[derive(Debug)]",
+            "enum ${1:Name} {",
+            "    ${2:Variant1},",
+            "    ${3:Variant2},",
+            "}"
+        ],
+        "description": "enum … { … }"
+    },
+    "extern-crate": {
+        "prefix": "extern-crate",
+        "body": [
+            "extern crate ${1:name};"
+        ],
+        "description": "extern crate …;"
+    },
+    "extern-fn": {
+        "prefix": "extern-fn",
+        "body": [
+            "extern \"C\" fn ${1:name}(${2:arg}: ${3:Type}) -> ${4:RetType} {",
```

Comment:
Should be `"extern \"C\" fn ${1:name}($2) $3 {"`

Reply:
even the `RetType` doesn't need a default name

Reply:
Ho yes sorry, I didn't understand your change proposition